File: KantechGatewayApp/Infrastructure/CsvUtil.vb
Imports System.IO
Imports System.Text
Namespace KantechGatewayApp.Infrastructure
    Public Module CsvUtil
        Public Function ReadRows(filePath As String, delimiter As Char, hasHeader As Boolean) As List(Of Dictionary(Of String, String))
            Dim rows As New List(Of Dictionary(Of String, String))()
            Dim headers() As String = Nothing
            Using sr As New StreamReader(filePath, Encoding.UTF8, True)
                Dim line As String = sr.ReadLine()
                If line Is Nothing Then Return rows
                Dim first = SplitCsv(line, delimiter)
                If hasHeader Then
                    headers = first
                Else
                    headers = first.Select(Function(x, i) $"Col{i + 1}").ToArray()
                    rows.Add(ToRow(headers, first))
                End If
                While True
                    line = sr.ReadLine()
                    If line Is Nothing Then Exit While
                    Dim parts = SplitCsv(line, delimiter)
                    rows.Add(ToRow(headers, parts))
                End While
            End Using
            Return rows
        End Function

        Private Function ToRow(headers() As String, parts() As String) As Dictionary(Of String, String)
            Dim d As New Dictionary(Of String, String)(StringComparer.OrdinalIgnoreCase)
            For i = 0 To headers.Length - 1
                Dim v = If(i < parts.Length, parts(i), "")
                d(headers(i).Trim()) = v.Trim()
            Next
            Return d
        End Function

        ' Simple CSV split (no quoted-commas). Replace with a robust parser if needed.
        Private Function SplitCsv(line As String, delimiter As Char) As String()
            Return line.Split(delimiter)
        End Function
    End Module
End Namespace

File: KantechGatewayApp/Infrastructure/CsvUtil.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace KantechGatewayApp.KantechGatewayApp.Infrastructure
{
    public static class CsvUtil
    {
        public static List<Dictionary<string, string>> ReadRows(string filePath, char delimiter, bool hasHeader)
        {
            var rows = new List<Dictionary<string, string>>();
            string[] headers = null;
            using (var sr = new StreamReader(filePath, Encoding.UTF8, true))
            {
                string line = sr.ReadLine();
                if (line is null)
                    return rows;
                string[] first = SplitCsv(line, delimiter);
                if (hasHeader)
                {
                    headers = first;
                }
                else
                {
                    headers = first.Select((x, i) => $"Col{i + 1}").ToArray();
                    rows.Add(ToRow(headers, first));
                }
                while (true)
                {
                    line = sr.ReadLine();
                    if (line is null)
                        break;
                    string[] parts = SplitCsv(line, delimiter);
                    rows.Add(ToRow(headers, parts));
                }
            }
            return rows;
        }

        private static Dictionary<string, string> ToRow(string[] headers, string[] parts)
        {
            var d = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
            for (int i = 0, loopTo = headers.Length - 1; i <= loopTo; i++)
            {
                string v = i < parts.Length ? parts[i] : "";
                d[headers[i].Trim()] = v.Trim();
            }
            return d;
        }

        // Simple CSV split (no quoted-commas). Replace with a robust parser if needed.
        private static string[] SplitCsv(string line, char delimiter)
        {
            return line.Split(delimiter);
        }
    }
}